README.md

```diff
@@ -1 +1,516 @@
+    - `undefinedError`: the error type could not be defined
+  - **`ErrorMessageValues`**, is the expected value to be met, it will be `true` for required, and the minimum value expected for `minValue` for example.
+- `getObject()`: Returns the data field in the schema that this input refers to
+
+**Please notice that in all of the examples bellow it is assumed that the components are already children of a FormContext component**
+
+### useHidden(path)
+
+##### Description
+
+Use this hook to build a hidden field in the form, the user will not be able to change it or see it, but it will be there when submitted.
+
+##### Parameters:
+
+- `path`: String which represents the path to the data field of the JSON Schema that this input will be built for.
+
+##### Return:
+
+Returns an object with the following fields, besides the common one's:
+
+- `getLabelProps()`: use this with the spread operator inside a `<label>` tag and get the benefit of having all the important fields of the label filled in for you and the associated input (the `for` property) with it.
+- `getInputProps()`: use this with the spread operator inside an `<input>` tag and get the benefit of the validator, id field, name and an associated label with it
+
+##### Example:
+
+```JSX
+function HiddenField(props) {
+  const inputMethods = useHidden('#/some/child/you/want/hidden');
+
+  return (
+    <React.Fragment>
+      <label {...inputMethods.getLabelProps()}>
+        {inputMethods.name}
+      </label>
+      <input {...inputMethods.getInputProps()} />
+    </React.Fragment>
+  )
+}
+```
+
+### useInput(path)
+
+##### Description
+
+Use this hook to build a generic input field in your form, with validation based on the type of input the JSON Schema requires.
+
+##### Parameters:
+
+- `path`: String which represents the path to the data field of the JSON Schema that this input will be built for.
+
+##### Return:
+
+Returns an object with the following fields, besides the common one's:
+
+- `getLabelProps()`: use this with the spread operator inside a `<label>` tag and get the benefit of having all the important fields of the label filled in for you and the associated input (the `for` property) with it.
+- `getInputProps()`: use this with the spread operator inside an `<input>` tag and get the benefit of the validator, id field, name and an associated label with it.
+
+##### Example:
+
+```JSX
+function InputField(props) {
+  const inputMethods = useInput('#/some/child/you/want/as/input');
+
+  return (
+    <React.Fragment>
+      <label {...inputMethods.getLabelProps()}>
+        {inputMethods.name}
+      </label>
+      <input {...inputMethods.getInputProps()} />
+    </React.Fragment>
+  )
+}
+```
+
+### useObject(path, UISchema)
+
+##### Description
+
+This hook works a little differently than the others. This hook will return an array of which each of its elements corresponds to the return type of one of the other hooks.
```

Comment:
To prevent repeating `this hook...`

```suggestion
This hook works a little differently than the others. `useObject` returns an array of which each of its elements corresponds to the return type of one of the other hooks.
```